AnySerial.cpp

```diff
@@ -373,7 +373,7 @@ AnySerial::library_version() {
 #endif
 #ifdef SoftwareSerial_h
         case anyserial_soft:
-            ret = serialport.soft->library_version();
+            // not implemented
```

Comment:
I don't have the code in front of me, but looks like there is a static class method to query the version.

Reply:
This might be a problem with me using the super-outdated Arduino IDE 1.0.5 from the Ubuntu repositories. The `SoftwareSerial`-library doesn't seem to have this method in this version.

Reply:
No, the Arduino IDE 1.0.5's `SoftwareSerial` does not include this method:

```bash
locate SoftwareSerial.h | xargs grep -q library_version && echo "found" || echo "not found"                                                                                                                                             
# not found
```

Reply:
Also in the Arduino repository the file https://github.com/arduino/Arduino/blob/master/hardware/arduino/avr/libraries/SoftwareSerial/src/SoftwareSerial.cpp does not mention any `library_version` method.

Reply:
This simple sketch

```c++
#include <SoftwareSerial.h>
#include <AnySerial.h>

SoftwareSerial SoftSerial(10,11);

void setup()
{
  AnySerial *anyserial = new AnySerial(&SoftSerial);
}

void loop() {}
```

fails with the up-to-date Arduino IDE 1.8.5:

```
Arduino: 1.8.5 (Linux), Board: "Arduino/Genuino Mega or Mega 2560, ATmega2560 (Mega 2560)"

/tmp/ccGWKRcD.ltrans0.ltrans.o: In function `main':
ccGWKRcD.ltrans0.o:(.text.startup+0xea): undefined reference to `AnySerial::AnySerial(SoftwareSerial*)'
collect2: error: ld returned 1 exit status
exit status 1
Fehler beim Kompilieren für das Board Arduino/Genuino Mega or Mega 2560.

Dieser Bericht wäre detaillierter, wenn die Option
"Ausführliche Ausgabe während der Kompilierung"
in Datei -> Voreinstellungen aktiviert wäre.
```

Did this work for you previously?